agnpy/synchrotron/proton_synchrotron.py

```diff
@@ -0,0 +1,183 @@
+# module containing the synchrotron radiative process
+import numpy as np
+import astropy.units as u
+from astropy.constants import e, h, c, m_e, m_p, sigma_T, G
+from ..utils.math import axes_reshaper, gamma_e_to_integrate
+from ..utils.conversion import nu_to_epsilon_prime, B_to_cgs, lambda_c_p
+from .synchrotron import Synchrotron
+from .synchrotron import single_particle_synch_power, tau_to_attenuation
+
+__all__ = ["ProtonSynchrotron"]
+
+e = e.gauss
+B_cr = 4.414e13 * u.G  # critical magnetic field
+
+class ProtonSynchrotron:
+    """Class for synchrotron radiation computation
+
+    Parameters
+    ----------
+    blob : :class:`~agnpy.emission_region.Blob`
+        emitting region and electron distribution
+    ssa : bool
+        whether or not to consider synchrotron self absorption (SSA).
+        The absorption factor will be taken into account in
+        :func:`~agnpy.synchrotron.Synchrotron.com_sed_emissivity`, in order to be
+        propagated to :func:`~agnpy.synchrotron.Synchrotron.sed_luminosity` and
+        :func:`~agnpy.synchrotron.Synchrotron.sed_flux`.
+    integrator : func
+        function to be used for integration (default = `np.trapz`)
+	"""
+
+    def __init__(self, blob, ssa=False, integrator=np.trapz):
+        self.blob = blob
+        self.ssa = ssa
+        self.integrator = integrator
+
+    @staticmethod
+    def evaluate_tau_ssa(
+        nu,
+        z,
+        d_L,
+        delta_D,
+        B,
+        R_b,
+        n_p,
+        *args,
+        integrator=np.trapz,
+        gamma=gamma_e_to_integrate,
+    ):
+        """Computes the syncrotron self-absorption opacity for a general set
```

Comment:
synchrotron

Reply:
Thank you for the comments. The reason we put SSA was mostly because Cerruti also included it on his code for proton synchrotron (unless we interpreted wrong the plot from his code). Since it is not relevant, do you think we should exclude it for protons?

Reply:
if you mimick Matteo's code it is fine to keep it as it is, but you should include a reference to the original code in the docstring of ProtonSynchrotron class